doc/kernel/usermode/mpu_stack_objects.rst

```diff
@@ -0,0 +1,49 @@
+.. _mpu_stack_objects:
+
+MPU Stack Objects
+#################
+
+Thread Stack Creation
+=====================
+
+Thread stacks are declared staticly with :c:macro:`K_THREAD_STACK_DEFINE()`
+or embedded within structures using c:macro:`K_THREAD_STACK_MEMBER()`
+
+For architectures which utilize memory protection unit (MPU) hardware,
+stacks are physically contiguous allocations.  This contiguous allocation
+has implications for the placement of stacks in memory, as well as the
+implementation of other features such as stack protection and userspace.
+
+Stack Guards
+============
+
+Stack protection mechanisms require hardware support that can restrict access
+to memory.  Memory protection units can provide this kind of support.
+The MPU provides a fixed number of regions.  Each region contains information
+about the start, end, size, and access attributes to be enforced on that
+particuliar region.
+
+Stack guards are implemented by using a single MPU region and setting the
+attributes for that region to not allow write access.  If invalid accesses
+occur, a fault ensues.  The stack guard is defined at the bottom of the stack,
+or the lowest address.
+
+Memory Placement
+================
+
+During stack creation, a set of constraints are enforced on the allocation of
+memory.  These contraints include determining the alignment of the stack and
+the correct sizing of the stack.  During linking of the binary, these
+constraints are used to place the stacks properly.
+
+The main source of the memory constraints is the MPU design for the SoC.
+Some MPUs require that each region be aligned to a power of two.  These SoCs
+will have :option:`CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT` defined.
+This means that a 1500 byte stack should be aligned to a 2kB boundary.  This
+requires a power of two ceiling to be done on the alignment and size of the
+stack.  The result is that the actual memory allocation of a given stack may
+be larger than the request.
+
+For MPUs which do not have a power of two alignment constraint, the minimum
+alignment is required to be 32 bytes.
```

Comment:
is there a size implication here too, such as multiples of 32 bytes?

Reply:
No.  those implementations use a start/end address register.  The only requirement is that start/end are aligned on 32 bytes.

Reply:
This sentence probably needs to be a little more precise.